src/client/retry.rs

```diff
@@ -657,11 +664,15 @@ mod tests {
             );
         }
 
-        let e = do_request().await.unwrap_err().to_string();
+        let e = do_request().await.unwrap_err();
         assert!(
-            e.contains(" after 2 retries, max_retries: 2, retry_timeout: 1000s  - Server returned non-2xx status code: 502 Bad Gateway"),
+            e.to_string().contains(" after 2 retries, max_retries: 2, retry_timeout: 1000s  - Server returned non-2xx status code: 502 Bad Gateway"),
             "{e}"
         );
+        assert_eq!(
```

Comment:
I think it would help to add a comment here saying the use of `e.source()` was intentional

Perhaps something like

```suggestion
        // verify e.source() is available as well for users who need programatic access
        assert_eq!(
```

(also in the one below)